internal/controllers/topology/cluster/scope/state_test.go

```diff
@@ -115,7 +115,93 @@ func TestIsUpgrading(t *testing.T) {
 	}
 }
 
-func TestUpgrading(t *testing.T) {
+func TestMPIsUpgrading(t *testing.T) {
+	g := NewWithT(t)
+	scheme := runtime.NewScheme()
+	g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
+
+	tests := []struct {
+		name     string
+		mp       *clusterv1.MachinePool
+		machines []*clusterv1.Machine
```

Comment:
We will check the noderef's kubelet version instead of the machine's version. Thus, we'll need to add a list of `*corev1.Node` instead of `*clusterv1.Machine`. The logic for this is here: https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/topology/cluster/scope/state.go#L174